llvm/lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -2313,6 +2316,21 @@ Error BitcodeReader::parseAttributeGroupBlock() {
           i--;
 
           B.addConstantRangeAttr(Kind, MaybeCR.get());
+        } else if (Record[i] == 8) {
+          Attribute::AttrKind Kind;
+          if (Error Err = parseAttrKind(Record[++i], &Kind))
+            return Err;
+          if (!Attribute::isConstantRangeListAttrKind(Kind))
+            return error("Not a constant range list attribute");
+
+          ConstantRangeList CRL(64, false);
+          int RangeSize = Record[++i];
```

Comment:
Maybe assert i + (RangeSize * 2) < e
(please double check if I got that right)

Reply:
Done!

Reply:
each loop iter up to `RangeSize` will bump `i` twice, so should it be * 2 ?

Reply:
Oops, good catch! Done.